### PR TITLE
1 srikar gpio

### PR DESCRIPTION
**Pull request by Srikar55**
gpio

### CODE REVIEW

File: firmware/gpio/gpio.cpp [hunk 1]
```diff
@@ -0,0 +1,157 @@
+/**************************
```
Comment (sidd91) [inline firmware/gpio/gpio.cpp]: Please create a header file. So that we can all first agree on the header first.

File: firmware/gpio/gpio.cpp [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+};
+     enum class State{
+         lo=0,hi=1
+     };
+    Gpio(int port, int pin) : port_(port), pin_(pin) {
+        Pinsel();
+    }
+    void Pinsel()
+    {
+               // uint8_t pinReg = pin_ / 16;
```
Comment (sidd91) [inline firmware/gpio/gpio.cpp]: remove dead code

File: firmware/gpio/gpio.cpp [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+    else {
+        return nullptr;
+    }
+
+}
+
+};
+
+int main(void)
```
Comment (sidd91) [inline firmware/gpio/gpio.cpp]: The main needs to be moved into an example directory. Feel free to create an example directory and put the example there. 
@tarunreddy03 FYI

File: firmware/gpio/gpio.cpp [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+#include "LPC17xx.h"
+#include <cstdint>
+#include <iostream>
+#include <string>
+using namespace std;
+
+class Gpio{
+    public:
+    int port_,pin_;
```
Comment (sidd91) [inline firmware/gpio/gpio.cpp]: Why do the members need to be public? These can be made private.

File: firmware/gpio/gpio.cpp (hunk 1 above)
Comment (Srikar55) [reply firmware/gpio/gpio.cpp]: Created a header file

File: firmware/gpio/gpio.cpp (hunk 2 above)
Comment (Srikar55) [reply firmware/gpio/gpio.cpp]: Removed unnecessary code

File: firmware/gpio/gpio.cpp (hunk 3 above)
Comment (Srikar55) [reply firmware/gpio/gpio.cpp]: Created a separate header and main file

File: firmware/gpio/gpio.cpp (hunk 4 above)
Comment (Srikar55) [reply firmware/gpio/gpio.cpp]: Did not make the members private.

File: .vscode/c_cpp_properties.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (sidd91) [inline .vscode/c_cpp_properties.json:1]: Delete this file

File: .vscode/launch.json [hunk 6]
```diff
@@ -0,0 +1,24 @@
+{
```
Comment (sidd91) [inline .vscode/launch.json:1]: Delete

File: .vscode/settings.json [hunk 7]
```diff
@@ -0,0 +1,60 @@
+{
```
Comment (sidd91) [inline .vscode/settings.json:1]: Delete

**Pull request closed without merging**